### PR TITLE
K8SPSMDB-226 Helm chart for PSMDB operator

### PR DESCRIPTION
**Pull request by cap1984**

### CODE REVIEW

File: charts/psmdb-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          ports:
+          - containerPort: 60000
+            name: metrics
+          command:
+          - percona-server-mongodb-operator
+          env:
+            - name: WATCH_NAMESPACE
+              value: {{ default .Release.Namespace .Values.watchNamespace }}
+            - name: OPERATOR_NAME
+              value: {{ default "percona-server-mongodb-operator" .Values.operatorName }}
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/psmdb-operator/templates/deployment.yaml:37]: no objections

File: charts/psmdb-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+This is an implementation of PSMDB operator Deployment and other stuff could be found here:
+* <https://github.com/percona/percona-server-mongodb-operator>
+
+## Pre Requisites
+* Kubernetes 1.11+
+* PV support on underlying infrastructure (only if you are provisioning persistent volume).
+* Requires at least `v2.4.0` version of helm to support
+
+## Deloyment Details
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:11]:
```suggestion
## Deployment Details
```

File: charts/psmdb-operator/README.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
+# psmdb-operator: A chart for installing the Percona Kubernetes operator for MongoDB
+
+This is an implementation of PSMDB operator Deployment and other stuff could be found here:
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:3]:
```suggestion
This chart implements PSMDB operator deployment. The Operator itself can be found here:
```

File: charts/psmdb-operator/README.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+* <https://github.com/percona/percona-server-mongodb-operator>
+
+## Pre Requisites
+* Kubernetes 1.11+
+* PV support on underlying infrastructure (only if you are provisioning persistent volume).
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:8]:
```suggestion
* PV support on the underlying infrastructure - only if you are provisioning persistent volume(s).
```

File: charts/psmdb-operator/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+* Requires at least `v2.4.0` version of helm to support
+
+## Deloyment Details
+* <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/>
+
+## Chart Details
+This chart will do the following:
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:15]:
```suggestion
This chart will
```

File: charts/psmdb-operator/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+* Deploy PSMDB operator pod for the further MongoDB creation inside K8S
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:16]:
```suggestion
deploy a PSMDB Operator Pod for the further MongoDB creation in K8S.
```

File: charts/psmdb-operator/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+### Installing the Chart
+To install the chart with the release name `psmdb` using a dedicated namespace(recommended):
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:19]:
```suggestion
To install the chart with the `psmdb` release name using a dedicated namespace (recommended):
```

File: charts/psmdb-operator/README.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
+# psmdb-operator: A chart for installing the Percona Kubernetes operator for MongoDB
+
+This is an implementation of PSMDB operator Deployment and other stuff could be found here:
+* <https://github.com/percona/percona-server-mongodb-operator>
+
+## Pre Requisites
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:6]:
```suggestion
## Pre-requisites
```

File: charts/psmdb-operator/README.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+```sh
+helm repo add percona https://percona-lab.github.io/percona-helm-charts/
+helm install percona/psmdb-operator --version 0.1.9 --namespace my-namespace
+```
+
+The chart can be customized using the following configurable parameters:
+
+| Parameter                       | Description                                                                   | Default                                   |
+| ------------------------------- | ------------------------------------------------------------------------------| ------------------------------------------|
+| `image.repository`              | PSMDB operator container image name                                           | `percona/percona-server-mongodb-operator` |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:30]:
```suggestion
| `image.repository`              | PSMDB Operator container image name                                           | `percona/percona-server-mongodb-operator` |
```

File: charts/psmdb-operator/README.md [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+| `image.tag`                     | PSMDB operator container image tag                                            | `1.4.0`                                   |
+| `image.pullPolicy`              | PSMDB operator container pull policy                                          | `Always`                                  |
+| `image.pullSecrets`             | PSMDB operator pod pull secret                                                | `[]`                                      |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:33]:
```suggestion
| `image.pullSecrets`             | PSMDB Operator Pod pull secret                                                | `[]`                                      |
```

File: charts/psmdb-operator/README.md [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+| `replicaCount`                  | PSMDB operator pod quantity                                                   | `1`                                       |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:34]: It's better to start "Pod" and "Operator" words with a capital letter (like official K8S docs are doing)

File: charts/psmdb-operator/README.md [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+| `tolerations`                   | List of node taints to tolerate                                               | `[]`                                      |
+| `resources`                     | Resource requests and limits                                                  | `{}`                                      |
+| `nodeSelector`                  | Labels for pod assignment                                                     | `{}`                                      |
+| `watchNamespace`                | Set if you want to specify a namespace to watch                               | `""`                                      |
+| `nameOverride`                  | Set if you want to use a different operator name                              | `""`                                      |
+| `fullnameOverride`              | Set if DO NOT want to see chart name in the operators' name                   | `""`                                      |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:40]: It's  better to rephrase this line somehow to make it more clear

File: charts/psmdb-operator/README.md (hunk 2 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 3 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 4 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 5 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 6 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 7 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 8 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 9 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 10 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 11 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 12 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+| `image.pullPolicy`              | PSMDB Operator Container pull policy                                          | `Always`                                  |
+| `image.pullSecrets`             | PSMDB Operator Pod pull secret                                                | `[]`                                      |
+| `replicaCount`                  | PSMDB Operator Pod quantity                                                   | `1`                                       |
+| `tolerations`                   | List of node taints to tolerate                                               | `[]`                                      |
+| `resources`                     | Resource requests and limits                                                  | `{}`                                      |
+| `nodeSelector`                  | Labels for Pod assignment                                                     | `{}`                                      |
+| `watchNamespace`                | Set when a different from default namespace is needed to watch                | `""`                                      |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:38]: Is this option officially supported?

File: charts/psmdb-operator/README.md [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+| `nameOverride`                  | Set if the chart name is not desired to be the operators name                 | `""`                                      |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:39]:
```suggestion
```
It looks like this option is not officially supported

File: charts/psmdb-operator/README.md [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+| `fullnameOverride`              | By default operator name will contain chart name concatenated with operator name.<br>Set this variable if you want to set the whole bunch of names  | `""`                                      |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:40]:
```suggestion
```
It looks like this option is not officially supported

File: charts/psmdb-operator/README.md [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+| `createCRD`                     | Set to false if you don't want the helm chart to automatically create the CRD | `true`                                    |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:41]:
```suggestion
```
It looks like this option is not officially supported

File: charts/psmdb-operator/README.md [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+| `env.resyncPeriod`              | Period for syncing operator with DB                                           | `5s`                                      |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:42]:
```suggestion
```

File: charts/psmdb-operator/README.md [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+| `env.logVerbose`                | Enable/Disable operator verbose logging                                       | `false`                                   |
```
Comment (fiowro) [inline charts/psmdb-operator/README.md:43]:
```suggestion
```

File: charts/psmdb-operator/README.md (hunk 14 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 15 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 16 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 17 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

File: charts/psmdb-operator/README.md (hunk 18 above)
Comment (cap1984) [reply charts/psmdb-operator/README.md]: done

**Review by fiowro — APPROVED** [review]
lgtm)

File: charts/psmdb-operator/Chart.yaml [hunk 19]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+appVersion: "1.4.0"
```
Comment (tplavcic) [inline charts/psmdb-operator/Chart.yaml:2]: Do we want 1.5.0 here?

File: charts/psmdb-operator/crds/crd.yaml [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+      served: true
+    - name: v1-2-0
+      storage: false
+      served: true
+    - name: v1-3-0
+      storage: false
+      served: true
+    - name: v1-4-0
+      storage: false
+      served: true
```
Comment (tplavcic) [inline charts/psmdb-operator/crds/crd.yaml:30]: Probably needs 1.5.0 added also

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: charts/psmdb-operator/values.yaml [hunk 21]
```diff
@@ -0,0 +1,48 @@
+# Default values for psmdb-operator.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+
+image:
+  repository: percona/percona-server-mongodb-operator
+  tag: 1.4.0
```
[the remark this answers is not in the document]
Comment (cap1984) [reply charts/psmdb-operator/values.yaml]: done

File: charts/psmdb-operator/Chart.yaml (hunk 19 above)
Comment (cap1984) [reply charts/psmdb-operator/Chart.yaml]: done

File: charts/psmdb-operator/crds/crd.yaml (hunk 20 above)
Comment (cap1984) [reply charts/psmdb-operator/crds/crd.yaml:30]: done

**Review by tplavcic — APPROVED** [review]

**Pull request merged**